Add tests for ansible and system manager modules

jfroche opened the pull request:
We are slowly moving ansible code to nix system manager modules.
To speed up the migration, we want to test the ansible code without building the whole AMI and have a fast feedback loop.
We also want to quickly test our changes on system manager modules.

The tests are running docker images with ubuntu using testinfra.

jfroche committed:
chore: update (and refactor) flake inputs

jfroche committed:
chore: replace (deprecated) `substituteAll` by `replaceVars`

jfroche committed:
chore: fix missing required `teams` meta attribute

jfroche committed:
chore: update old (unsupported) version of nix used in script

jfroche committed:
fix: overlay and use our `buildPgrxExtension` function

These overlay packages will be removed once the extensions are updated
to use the new `buildPgrxExtension` function.

jfroche committed:
chore: fix formatting

yvan-sraka committed:
chore: bump `pgrouting` version to 3.8.0

The previously used version was failing to build:

/build/source/src/common/get_check_data.c: In function 'pgr_SPI_getText':
/build/source/src/common/get_check_data.c:307:28: error: passing argument 1 of 'DatumGetCString' makes integer from pointer without a cast [-Wint-conversion]
  307 |     return DatumGetCString(SPI_getvalue(*tuple, *tupdesc, info.colNumber));
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                            |
      |                            char *
In file included from /build/source/include/c_common/postgres_connection.h:29,
                 from /build/source/src/common/get_check_data.c:27:
/nix/store/nn6vpjqlsxxmynwcrsib9agb3hpr5cqj-postgresql-17.4/include/server/postgres.h:335:23: note: expected 'Datum' {aka 'long unsigned int'} but argument is of type 'char *'
  335 | DatumGetCString(Datum X)
      |                 ~~~~~~^
make[2]: *** [src/common/CMakeFiles/common.dir/build.make:149: src/common/CMakeFiles/common.dir/get_check_data.c.o] Error 1

jfroche committed:
fix: backport tzdata fixes for postgresql 15

This was discussed upstream here:
https://postgr.es/m/0a997455-5aba-4cf2-a354-d26d8bcbfae6%40technowledgy.de

Also applied by nixpkgs here: https://github.com/NixOS/nixpkgs/pull/342026

jfroche committed:
fix: plv8 3.1 requires older v8

We cannot use nodejs.libv8 as it is too new for plv8 3.1.
We will be able to use it with plv8 3.2.

jfroche committed:
chore: update cargo-pgrx 1.14.3 cargo hash

jfroche committed:
chore(postgresql): add isOrioleDB condition

jfroche committed:
fix: pgregress tests

jfroche committed:
fix: update schema for dbmate schema_migrations table

dbmate 2.27 is now using a varchar for the version column in the
schema_migrations table.

See https://github.com/amacneil/dbmate/pull/641

jfroche committed:
feat: add ansible task testing infrastructure based on Docker and pytest

This complements the existing AMI tests in testinfra by providing
a faster feedback loops for Ansible development without requiring a full
VM.

Also as it is based on Docker, it can be run locally (e.g. macOS) or in CI.

Note that this approach is not intended to replace the AMI tests, but
rather to provide a more efficient way to test Ansible tasks during
development.

Docker is used outside of the Nix sandbox for the moment.

You can run the tests using `nix run -L .\#ansible-test`

jfroche committed:
feat: deploy nginx using system manager

And use docker to run tests

jfroche committed:
chore: update fmt

jfroche committed:
Structured attributes are now required when building with separateDebugInfo

See https://github.com/NixOS/nixpkgs/pull/394674

jfroche committed:
chore: simplify system manager tests

Move ubuntu docker image to its own package.
Create one test for all system modules.

jfroche committed:
chore: formatting

jfroche committed:
Test system manager deployment triggered by Ansible

jfroche committed:
Create default system manager configuration

jfroche committed:
ci: make sure we cache system manager and its test dependencies

samrose committed:
chore: version bump

hunleyd commented on ansible/tasks/setup-nix.yml:
```diff
@@ -0,0 +1,11 @@
+---
+- name: Check if nix is installed
+  ansible.builtin.command: which nix
+  register: nix_installed
+  failed_when: nix_installed.rc != 0
+  ignore_errors: true
+
+- name: Install nix
+  ansible.builtin.shell: curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install --no-confirm --extra-conf 'substituters = https://cache.nixos.org https://nix-postgres-artifacts.s3.amazonaws.com' --extra-conf 'trusted-public-keys = nix-postgres-artifacts:dGZlQOvKcNEjvT7QEAJbcV6b6uk7VF/hWMjhYleiaLI=% cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGsp ...
+  when: nix_installed.rc != 0
+  become: true
```
instead of `ignore_errors`, why not use a block/rescue
```suggestion
- name: Install nix if needed
  block:
    - name: Check if nix is installed
      ansible.builtin.command: which nix
      register: nix_installed
      failed_when: nix_installed.rc != 0
  rescue:
    - name: Install nix
      ansible.builtin.shell: curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install --no-confirm --extra-conf 'substituters = https://cache.nixos.org https://nix-postgres-artifacts.s3.amazonaws.com' --extra-conf 'trusted-public-keys = nix-postgres-artifacts:dGZlQOvKcNEjvT7QEAJbcV6b6uk7VF/hWMjhYleiaLI=% cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY='
      become: true
```

samrose committed:
fix: rebase on develop and fix code api changes from nixpkgs

samrose committed:
tests: try to get passing tests

samrose committed:
feat: pin icu75 for postgres build to maintain compat with existing
deployments

samrose committed:
chore: remove commented out code not needed in latest changes to pg

yvan-sraka committed:
chores: add nix run .#ansible-test and .#check-system-module to github actions workflows

yvan-sraka committed:
fix: apparently incorrectly rebased changes, and % nix fmt

yvan-sraka committed:
fix: replace deprecated v8 package, and wrong packages arguments

yvan-sraka committed:
chore: remove nixpkgs-pgbackrest input, and % nix flake update

jfroche committed:
chore: fix formatting

jfroche committed:
fix(darwin): remove reference to apple-sdk

jfroche committed:
fix(ext): missing required attributes in some extension tests

yvan-sraka committed:
fix(postgis): remove custom sfcgal package dependency

Updated postgis.nix to use nixpkgs sfcgal instead of our custom package,
eliminating the need to maintain a separate sfcgal build definition.

yvan-sraka committed:
fix(pg_net): pin curl to older nixpkgs version

pg_net requires an older version of curl (pinned from nixpkgs commit
a76c4553d7e741e17f289224eda135423de0491d) to resolve build compatibility issues.

yvan-sraka committed:
fix(pgrx): error: multiple input filenames provided

Updated cargo-pgrx buildPgrxExtension with rustc wrapper to handle empty
postmaster_stub.rs files that cause duplicate symbol errors.
(first two filenames are `` and
`/build/tmp.*/postmaster_stubs/nix/store/*/bin/postmaster_stub.rs`)

pg_jsonschema cargo checks fails on specific versions (0.2.0, 0.3.1, 0.3.3).

yvan-sraka committed:
chore: % nix flake update && nix fmt

Updated nixpkgs, rust-overlay, and nix-fast-build to latest versions.

yvan-sraka committed:
fix(orioledb): fixed installedExtensions reference to use proper self.packages

yvan-sraka committed:
fix(postgresql): remove redundant patch substitutions

Removed duplicate substituteInPlace calls that were redundantly applied during previous merge operations. Cleaned up the patch application to avoid duplicate substitutions.

yvan-sraka committed:
fix(tests): updated test expectations

Remove deprecated functions that were removed in newer pgrouting versions
(_pgr_betweennesscentrality and _pgr_contractionhierarchies).

yvan-sraka committed:
feat: (re)add check_pg_regress method to PostgresExtensionTest

yvan-sraka committed:
fix(pgrx): remove deprecated darwin.apple_sdk.frameworks.Security

yvan-sraka committed:
refactor: replace fetchTarball with flake input for nixpkgs-oldstable

Improves reproducibility and avoids evaluation-time network access by
using a proper flake input instead of fetchTarball for the older nixpkgs
needed by pg_net and plv8.

yvan-sraka committed:
fix(pg_net): disable dangling-assignment warning for v0.19.5 on aarch64-darwin

Version 0.19.5 has a dangling pointer issue in src/core.c:177 that causes
compilation to fail on aarch64-darwin with newer clang versions. This adds
-Wno-error=dangling-assignment to allow compilation to proceed.

The flag is restricted to aarch64-darwin only since GCC on Linux doesn't
recognize this warning option.

yvan-sraka committed:
fix(plv8): add headerpad_max_install_names for macOS builds

Add -headerpad_max_install_names linker flag to prevent 'install_name_tool:
changing install names or rpaths can't be redone' errors on macOS. This
allocates sufficient header space for install_name_tool to modify the
shared library during post-install processing.

yvan-sraka committed:
fix(tests): correct library naming and symlinks for background worker extensions

Test assertions were inconsistent: expecting {version}.so and {extension_name}-{version}.so
patterns while actual libraries use {lib_name}-{version}.so (e.g. pg_partman_bgw-5.3.1.so,
timescaledb-loader-2.16.1.so). Updated PostgresExtensionTest to accept Optional[str] lib_name
parameter and use consistent {lib_name}-{version}.so assertions throughout.

Added missing timescaledb-loader.so symlink creation and updated the switch script to
maintain both symlinks when switching versions.

Fixes 'Expected timescaledb version 2.16.1, but found timescaledb-loader-2.16.1.so' and
'Expected pg_partman version 5.3.1, but found pg_partman_bgw-5.3.1.so' test failures.

yvan-sraka committed:
refactor(exts): replace nixpkgs-oldstable with specific overlay packages

Use overlay instead of passing full nixpkgs-oldstable to avoid exposing
entire old nixpkgs attribute set. Now provides specific packages
(curl_8_4, v8_oldstable) through overlay.

yvan-sraka committed:
fix: restore revision support for orioledb builds (introduced it #1880)

Keep revision support that was accidentally simplified during nixpkgs update.
Restores conditional logic to support building orioledb from specific git
revision hashes instead of only tags.

yvan-sraka committed:
fix(exts): remove old files that were moved to subdirectories

These files were moved to subdirectories and the old .nix files should be
removed since they were moved to new directory structure with default.nix files.

yvan-sraka committed:
fix(flake): remove nixpkgs-go124 and nix-fast-build dependencies

Remove nixpkgs-go124 dependency since current nixpkgs has Go 1.25.4
which is sufficient for packer. Remove nix-fast-build input as unused
and add follows directives to resolve duplicate dependencies.

yvan-sraka committed:
fix(flake): update flake.lock and cargo-pgrx hashes

Run flake-linter to eliminate duplicate dependencies, and remove unused
cargo-pgrx / rustc versions hashes.

yvan-sraka committed:
fix(exts): remove remaining duplicated files

Move pgrouting.nix to pgrouting/default.nix and groonga package to
pgroonga/groonga.nix. Update imports and fix relative paths for versions.json
and patch files.

yvan-sraka committed:
refactor(pgrx): conditionally apply rustc wrapper when < 0.12

Replace inline bash script generation with writeShellScriptBin for rustc
wrapper that filters empty postmaster_stub.rs arguments. Apply wrapper only
for pgrx < 0.12 since issue was fixed upstream in pgrx#1435 and pgrx#1441.

yvan-sraka committed:
refactor(nixpkgs): use let binding for oldstable instead of overlay

Move nixpkgs-oldstable import to a let binding to avoid importing it
three times...

yvan-sraka committed:
refactor(groonga): move groonga package to pgroonga extension

Remove supabase-groonga from global flake packages and import it locally
in pgroonga extension instead. Other components access groonga via
pgroonga.passthru.groonga since groonga is only used by pgroonga.

yvan-sraka committed:
chore: bump version suffix to test

yvan-sraka committed:
fix(ansible): remove supabase-groonga from package install list

The supabase-groonga package was moved to pgroonga extension in commit 1f0ed1f0,
but Ansible was still trying to install it separately causing build failures.

yvan-sraka committed:
Revert "fix(ansible): remove supabase-groonga from package install list"

This reverts commit 55474686405394826ad07a032cbcf0e36a924554.

yvan-sraka committed:
Revert "refactor(groonga): move groonga package to pgroonga extension"

This reverts commit cfa9bcb92219f391469ae743f95c8856b0a99561.

yvan-sraka committed:
fix(darwin): remove deprecated apple_sdk frameworks from supascan

Remove references to darwin.apple_sdk.frameworks.{IOKit,CoreFoundation}
which have been deprecated in nixpkgs. Disable CGO to avoid Darwin
framework dependencies entirely.

samrose committed:
chore: bump to retest

samrose committed:
chore: suffix to test

samrose committed:
chore: bump to test

jfroche commented:
Superseded by #1882 and #2010
